src/Elastic.Apm/Report/Serialization/BodyStringSanitizerConverter.cs

```diff
@@ -59,6 +59,9 @@ public override void WriteJson(JsonWriter writer, object value, JsonSerializer s
 		/// <returns></returns>
 		private static bool FormatCheck(string bodyValue)
 		{
+			if (!bodyValue.Contains("=") && !bodyValue.Contains("&"))
```

Comment:
It seems to be very inefficient - with this change in the worst case `WriteJson` will iterate over the body 4 times.

Reply:
How about https://github.com/elastic/apm-agent-dotnet/pull/617/commits/46e97f06851f097cbfa660e9d487a89a9b371c21? 

Overall this whole thing is not efficient - reading the body will add lots of overhead, and returning and parsing the keys adds even more I think. I feel what I talk about [here](https://github.com/elastic/apm-agent-dotnet/pull/617#issuecomment-557001669) could make a real perf. difference. 

This discussion is yet another reason to finish https://github.com/elastic/apm-agent-dotnet/issues/28 :)

Reply:
Please see - https://github.com/elastic/apm-agent-dotnet/pull/617/files#r349052916
I don't think benchmarks are necessary to avoid iterating multiple times over potentially quite a big buffer.

Reply:
> I don't think benchmarks are necessary to avoid iterating multiple times over potentially quite a big buffer.

:) Sure - benchmarks don't avoid iterations, they deliver data. 

What I meant is that benchmarks are useful to discuss performance. Discussing performance without benchmark is what I don't like doing.